packages/console/src/pages/ApplicationDetails/components/Permissions/ApplicationScopesManagementModal/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { ApplicationUserConsentScopeType } from '@logto/schemas';
+import { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+
+import Button from '@/ds-components/Button';
+import DangerousRaw from '@/ds-components/DangerousRaw';
+import FormField from '@/ds-components/FormField';
+import ModalLayout from '@/ds-components/ModalLayout';
+import TextInput from '@/ds-components/TextInput';
+import useActionTranslation from '@/hooks/use-action-translation';
+import useApi from '@/hooks/use-api';
+import * as modalStyles from '@/scss/modal.module.scss';
+import { trySubmitSafe } from '@/utils/form';
+
+import { type ScopesTableRowDataType, type UserScopeTableRowDataType } from '../use-scopes-table';
+
+/**
+ * ApplicationScopesManagementModal
+ *
+ * This modal is used to edit the resource and organization scopes' descriptions.
+ */
+export type EditableScopeData = Exclude<ScopesTableRowDataType, UserScopeTableRowDataType>;
+
+type Props = {
+  scope?: EditableScopeData;
+  onClose: () => void;
+  onSubmit: (scope: EditableScopeData) => void;
+};
+
+function ApplicationScopesManagementModal({ scope, onClose, onSubmit }: Props) {
+  const api = useApi();
+
+  const {
+    reset,
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<EditableScopeData>();
+
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const tAction = useActionTranslation();
+
+  const onSubmitHandler = handleSubmit(trySubmitSafe(onSubmit));
+
+  // Reset form on open
+  useEffect(() => {
+    if (scope) {
+      reset(scope);
+    }
+  }, [reset, scope]);
```

Comment:
How about use the `scope` as the default value of the `useForm` hook? then we can remove this useEffect.

Reply:
default value won't update with the input props.